src/Sulu/Bundle/AudienceTargetingBundle/Mapper/TargetGroupMapper.php

```diff
@@ -0,0 +1,137 @@
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function mapDataToTargetGroup(TargetGroupInterface $targetGroup, array $data, $doPatch = true)
+    {
+        $targetGroup->setTitle($this->getConditionalValue($data, 'title', null, $targetGroup->getTitle(), $doPatch));
+        $targetGroup->setPriority(
+            $this->getConditionalValue($data, 'priority', null, $targetGroup->getPriority(), $doPatch)
+        );
+        $targetGroup->setActive((bool) json_decode($this->getValue($data, 'active', $targetGroup->isActive())));
+        $targetGroup->setDescription(
+            $this->getConditionalValue($data, 'description', null, $targetGroup->getDescription(), $doPatch)
+        );
+        $targetGroup->setAllWebspaces(
+            (bool) json_decode($this->getValue($data, 'allWebspaces', $targetGroup->isAllWebspaces()))
+        );
+
+        $webspaces = $this->getConditionalValue($data, 'webspaces', [], false, $doPatch);
+        if ($webspaces !== false) {
+            $this->mapWebspaceData($targetGroup, $webspaces);
+        }
+    }
+
+    /**
+     * Maps webspace data to given target group.
+     *
+     * @param TargetGroupInterface $targetGroup
+     * @param array $webspaceData
+     */
+    protected function mapWebspaceData(TargetGroupInterface $targetGroup, array $webspaceData)
+    {
+        // Remove all webspaces from targetGroup are not given.
+        $webspaceKeys = array_column($webspaceData, 'webspaceKey');
+        foreach ($targetGroup->getWebspaces() as $targetGroupWebspace) {
+            if (!in_array($targetGroupWebspace->getWebspaceKey(), $webspaceKeys)) {
+                $this->targetGroupWebspaceRepository->remove($targetGroupWebspace);
+                $targetGroup->removeWebspace($targetGroupWebspace);
+            }
+        }
+
+        foreach ($webspaceData as $data) {
+            $webspaceKey = $this->getValue($data, 'webspaceKey', null);
+            $targetGroupWebspace = $this->targetGroupWebspaceRepository->findOrCreate($targetGroup, $webspaceKey);
+            // Add to webspaces if not already added.
+            if (!$targetGroup->getWebspaces()->contains($targetGroupWebspace)) {
+                $targetGroup->addWebspace($targetGroupWebspace);
+            }
+            $targetGroupWebspace->setWebspaceKey($webspaceKey);
+            $targetGroupWebspace->setTargetGroup($targetGroup);
+        }
+    }
+
+    /**
+     * Returns property of data with given name.
+     * If this property does not exists this function returns given default
+     * based on if patch is true or false.
+     *
+     * @param array $data
+     * @param string $name
+     * @param mixed|null $putDefault
+     * @param mixed|null $patchDefault
+     * @param bool $doPatch
+     *
+     * @return mixed
+     */
+    protected function getConditionalValue(
+        array $data,
+        $name,
+        $putDefault = null,
+        $patchDefault = null,
```

Comment:
So that i can handle both, put and patch in one single function. It's a helper to make the mapping function more compact. Here's an example what it would look like without this function.


```
$default = 'putDefault';
if ($doPath) {
    $default = 'patchDefault';
}
$example->set($this->getValue($data, 'my-property', $default));
```

I would need to do this for each property i want to set. Now i just need to type:

```
$example->set($this->getConditionalValue($data, 'my-property', 'putDefault', 'patchDefault', $doPatch));
```

that's the whole point